geps/gep-2643/index.md

```diff
@@ -0,0 +1,592 @@
+* [Ana] realizes she could use TCPRoute, but then she would need different ports for each service on the listener (because `TCPRoute` traffic is classified only by the dstip:dstport)
+* [Ana] figures out she can use the `SNI attribute` to listen to the traffic all on the same port (5432) but classify it based on the SNI attribute. This way the same port can be used, but given this is a TLS traffic, the requested hostname would be used to route the traffic to the right Postgres instance.
+
+[Ana]: https://gateway-api.sigs.k8s.io/concepts/roles-and-personas/#ana
+
+
+## API
+
+```golang
+// TLSRouteSpec defines the expected behavior of a TLSRoute.
+// A TLSRoute MUST be attached to a Listener of protocol TLS.
+// Core: The listener CAN be of type Passthrough
+// Extended: The listener CAN be of type Terminate 
+type TLSRouteSpec struct {
+  CommonRouteSpec `json:",inline"`
+  
+  // Hostnames defines a set of SNI hostnames that should match against the
+  // SNI attribute of TLS ClientHello message in TLS handshake. This matches
+  // the RFC 1123 definition of a hostname with 2 notable exceptions:
+  //
+  // 1. IPs are not allowed in SNI hostnames per RFC 6066.
+  // 2. A hostname may be prefixed with a wildcard label (`*.`). The wildcard
+  //    label must appear by itself as the first label.
+  //
+  // <gateway:util:excludeFromCRD>
+  // If a hostname is specified by both the Listener and TLSRoute, there
+  // must be at least one intersecting hostname for the TLSRoute to be
+  // attached to the Listener. For example:
+  //
+  // * A Listener with `test.example.com` as the hostname matches TLSRoutes
+  //   that have specified at least one of `test.example.com` or
+  //   `*.example.com`.
+  // * A Listener with `*.example.com` as the hostname matches TLSRoutes
+  //   that have specified at least one hostname that matches the Listener
+  //   hostname. For example, `test.example.com` and `*.example.com` would both
+  //   match. On the other hand, `example.com` and `test.example.net` would not
+  //   match.
+  // * A listener with `something.example.com` as the hostname matches a 
+  //   TLSRoute with hostname `*.example.com`.
+  //
+  // If both the Listener and TLSRoute have specified hostnames, any
+  // TLSRoute hostnames that do not match any Listener hostname MUST be
+  // ignored. For example, if a Listener specified `*.example.com`, and the
+  // TLSRoute specified `test.example.com` and `test.example.net`,
+  // `test.example.net` must not be considered for a match.
+  //
+  // If both the Listener and TLSRoute have specified hostnames, and none
+  // match with the criteria above, then the TLSRoute is not accepted for that 
+  // Listener. If the TLSRoute does not match any Listener on its parent, the 
+  // implementation must raise an 'Accepted' Condition with a status of
+  // `False` in the corresponding RouteParentStatus.
+  // 
+  // A Listener MUST be of type TLS when a TLSRoute attaches to it. The 
```

Comment:
```suggestion
  // A Listener MUST be have protocol set to TLS when a TLSRoute attaches to it. The 
```
It's a small thing, but the field is not called `type` - we have to be as precise as possible with this.